### ISSUE TITLE
Sketch out desired features

### ISSUE THREAD

**Issue by jacobpennington** [jacobpennington/nems-lite#1]
Plan out the big things we want in the new version. Turn these into separate "enhancement" issues after list is finalized.

1) Simple API for fitting models that does not rely on xforms, keywords, cellid/batch, etc. Just assume data has been loaded as a generic format (probably a 2D NumPy array). Use scikit-learn for inspiration on API style. Something like:
```
stimulus, response = my_data_loader('/path/to/my_data.csv')
model = ModelSpec(...)  # specify modules somehow, TBD
model.fit(stimulus, response, ...)
```
with reasonable defaults hidden from novice users but a simple way to specify detailed options when needed.
    ~~1a) Recordings behind the scenes. I think I already wrote an array -> Recording utility.~~
    1b) Similar API that takes in a ~~recording~~dictionary, for people that have more complicated models
          (ex: feeding in both stim and state)

2) New class-based ~~Module~~Layer system, integrate into updated Model~~Spec~~ class.
    2a) Option to specify keywords for ~~modules~~Layers still, but not separate as before,
    2b) Scheme for specifying different operation based on backend (TF, scipy, ...)
          Ex:ample:
          `Layer.fittable(backend='scipy')`
          returns all parameters as a vector, but
          `Layer.fittable(backend='tf')`
          returns an appropriately parameterized Tensorflow layer.

3) Helper function(s) to map old NEMS models to new system for backwards compatibility.
Keep this compartmentalized! Design new features based on what we want to do, not what
we did before.
    3a) Should only need to convert the model. Preprocessing etc. would happen outside the new
          fit API, and a lot of that stuff is lab specific. So xforms can just continue as part of nems_lbhb if desired.

4) Forward compatibility with complicated fit routines. Getting things like the gc model to work
through xforms was a pain. Try to keep flexibility or alternate fit strategies like that in mind
when designing, instead of hacking together afterward.
    4a) Simplest solution: just keep those details out of NEMS entirely, leave it up to the modeler to figure that
          out in their own scripts.

5) (maybe) Switch to xarrays for signal data? Other places too?
6) make time axis 0 for signals.
